accepted/00062-hub-xmlrpc-api.md

```diff
@@ -0,0 +1,129 @@
+- Feature Name: hub_xmlrpc_api
+- Start Date: 2019-10-10
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Uyuni Server has an XMLRPC API that allows access to most of its functionality.
+
+With the Hub architecture, there will be multiple Servers - this RFC details a new XMLRPC API for the Hub to access Server functionality.
+
+See the [Hub general RFC](accepted/00060-hub-general.md) for an introduction to the Hub project.
+
+# Motivation
+[motivation]: #motivation
+
+End users might not want to access single Servers to automate operations across their infrastructure, so a Hub XMLRPC API endpoint would be useful to make third party scripting and integration easier.
+
+From the requirements:
+
+> Hub implements API endpoints to act on SUMA slave Servers
+>
+>    Main expected use cases:
+>        Content management (eg. channel cloning)
+>        Automated patching
+
+# Detailed design
+[design]: #detailed-design
+
+### Basic funtionality
+The Hub is a Server, so as far as it and its minions (Servers) are concerned, the Hub XMLRPC API work as of today.
+Content management APIs will work as of today. Inter-Server Synchronization (ISS) will be used to push managed content from Hub to Servers.
+
+### New service and endpoint
+A new XMLRPC API endpoint will be created, implemented by a new service called the "XMLRPC Gateway API" (simply called "Gateway" from now on in this document). Technology-wise:
+  * Implementation will be in Python based on top of Tornado and [tornado-xmlrpc](https://pypi.org/project/tornado-xmlrpc/) library
+  * all I/O should be handled asynchronously via Tornado. Calls to several Servers should happen in parallel, with a maximum timeout value for unreachable/not responding Servers
+  * there will be no backing database. All new methods will delegate to existing XMLRPC APIs (either the Hub's or individual Servers')
+
+### Topology exposure functionality
+Gateway methods will be created to expose how Servers, clients and Hubs are connected, in particular methods will be needed to:
+  * Get a list of `serverIds`
+  * Get a list of clients registered to a Server given its `serverId`
+  * Get a list of all clients, with the `serverIds` they belong to
+  * Get the `serverId` for a given client (based on `minion_id`, FQDN or other unique identifier. Normal client IDs will not work, as they will overlap between Servers)
```

Comment:
I understand and agree we have this problem, in general, especially about client-addressed routing (CAR), described in "Next steps" below.

For the mere topology exposure functionality, it will be OK if this particular method returns more than one `serverId`, adjusting text in this sense.